Clarify Enterprise required for multiple instances (#753)
mattermost/mattermost-plugin-jira: readme.md

This plugin supports a two-way integration between Mattermost and Jira. Jira Core and Jira Software products, for Server, Data Center, and Cloud platforms are supported. It has been tested with versions 7 and 8.

> From v3.0 of this plugin, support for multiple Jira instances is offered when using an Enterprise License.

Reviewer: ```suggestion
From v3.0 of this plugin, support for multiple Jira instances is offered for Mattermost Server Enterprise Edition and Mattermost Cloud users.
```

Reviewer: @jfrerich, is this correct?

Reviewer 2: I think you could leave it a `for Mattermost Enterprise Editions`

Author: ```suggestion
From v3.0 of this plugin, support for multiple Jira instances is offered for Mattermost Enterprise Editions.
```
